### ISSUE TITLE
Problem with disable pseudo terminal option, '-T'.

### ISSUE THREAD

**Issue by ggurgen** [PowerShell/Win32-OpenSSH#2085]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the latest version
- [X] Search the existing issues.

### Steps to reproduce

1. Update Windows 11.
2. Install openssh server from powershell or app>feature options.
    Installed version 8.6
3. From any linux client host try use ssh -T <user>@<Windows_host_IP>

### Expected behavior

```console
Successful login.
```

### Actual behavior

```console
<some linux host>:~$ ssh -T <user>@<IP> -vvv
***
debug3: send packet: type 98                                                          
debug3: Ignored env _                                                                 
debug2: channel 0: request shell confirm 1                                            
debug3: send packet: type 98                                                          
debug2: channel_input_open_confirmation: channel 0: callback done                     
debug2: channel 0: open confirm rwindow 0 rmax 32768                                  
debug3: send packet: type 1                                                           
packet_write_wait: Connection to <IP> port 22: Broken pipe
```

### Error details

```console
10368 2023-06-21 06:10:19.679 debug3: receive packet: type 98
10368 2023-06-21 06:10:19.679 debug1: server_input_channel_req: channel 0 request shell reply 1
10368 2023-06-21 06:10:19.679 debug1: session_by_channel: session 0 channel 0
10368 2023-06-21 06:10:19.679 debug1: session_input_channel_req: session 0 req shell
10368 2023-06-21 06:10:19.679 Starting session: shell for <user> from <IP> port 44676 id 0
10368 2023-06-21 06:10:19.679 debug2: fd 9 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 11 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 12 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 13 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 14 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 15 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug3: shell: "c:\\\\windows\\\\system32\\\\cmd.exe"
10368 2023-06-21 06:10:19.679 debug3: shell_option: /c
10368 2023-06-21 06:10:19.679 debug3: exec_command: (null)
10368 2023-06-21 06:10:19.679 debug3: arg escape option: TRUE
10368 2023-06-21 06:10:19.679 debug3: spawn_argv[0]: "c:\\\\windows\\\\system32\\\\cmd.exe"
10368 2023-06-21 06:10:21.695 debug3: mm_request_receive: entering
10368 2023-06-21 06:10:21.695 debug1: do_cleanup
10056 2023-06-21 06:10:21.695 debug2: pselect_notify_done: reading
```

### Environment data

```PowerShell
PS C:\Windows\system32> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.22000.2003
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.22000.2003
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Version

OpenSSH_for_Windows_8.9p1, LibreSSL 3.4.3

**Comment by ggurgen** [comment]
From version 8.6 till to 8.9.1 openssh server not open session if client side use '-T' option.
ssh -T user@win11host 
logs from client side looks:
***
debug3: send packet: type 98                                                          
debug3: Ignored env _                                                                 
debug2: channel 0: request shell confirm 1                                            
debug3: send packet: type 98                                                          
debug2: channel_input_open_confirmation: channel 0: callback done                     
debug2: channel 0: open confirm rwindow 0 rmax 32768                                  
debug3: send packet: type 1                                                           
packet_write_wait: Connection to <IP> port 22: Broken pipe 

From server side:
***
10368 2023-06-21 06:10:19.679 debug3: receive packet: type 98
10368 2023-06-21 06:10:19.679 debug1: server_input_channel_req: channel 0 request shell reply 1
10368 2023-06-21 06:10:19.679 debug1: session_by_channel: session 0 channel 0
10368 2023-06-21 06:10:19.679 debug1: session_input_channel_req: session 0 req shell
10368 2023-06-21 06:10:19.679 Starting session: shell for <user> from <IP> port 44676 id 0
10368 2023-06-21 06:10:19.679 debug2: fd 9 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 11 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 12 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 13 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 14 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug2: fd 15 setting O_NONBLOCK
10368 2023-06-21 06:10:19.679 debug3: shell: "c:\\\\windows\\\\system32\\\\cmd.exe"
10368 2023-06-21 06:10:19.679 debug3: shell_option: /c
10368 2023-06-21 06:10:19.679 debug3: exec_command: (null)
10368 2023-06-21 06:10:19.679 debug3: arg escape option: TRUE
10368 2023-06-21 06:10:19.679 debug3: spawn_argv[0]: "c:\\\\windows\\\\system32\\\\cmd.exe"
10368 2023-06-21 06:10:21.695 debug3: mm_request_receive: entering
10368 2023-06-21 06:10:21.695 debug1: do_cleanup
10056 2023-06-21 06:10:21.695 debug2: pselect_notify_done: reading

**Comment by tgauth** [comment]
This is a known issue - https://github.com/PowerShell/Win32-OpenSSH/issues/1993. Please use the latest GitHub release for the fix.

**Comment by ggurgen** [comment]
Would we wait fix back port to 8.6 version?

**Comment by tgauth** [comment]
Non-security fixes are not typically backported, unfortunately.
